xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/model/document/RequiredRights.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.model.document;
+
+import java.util.Set;
+
+import org.xwiki.security.authorization.Right;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Stores the required rights of a {@link org.xwiki.bridge.DocumentModelBridge}.
+ *
+ * @version $Id$
+ * @since 15.5RC1
+ */
+@Unstable
+public interface RequiredRights
+{
+    /**
+     * @return an immutable set of required rights
+     */
+    Set<Right> getRights();
+
+    /**
+     * @param newRights the new rights to set
+     */
+    void setRights(Set<Right> newRights);
+
+    /**
+     * @param right the right to check
+     * @return {@code true} of the right is part of the required rights, {@code false} otherwise
+     */
+    boolean has(Right right);
+
+    /**
+     * @return {@code true} when the required rights are activated for the document
```

Comment:
Would be to good to explain in the javadoc of the interface what's an "activated" right. It's not obvious when reading it.